### ISSUE TITLE
Model Building

### ISSUE THREAD

**Issue by aelaraby6** [aelaraby6/car-price-prediction#3]
## 🤖 Tasks

- [ ] Load preprocessed data
- [ ] Try multiple models:
- [ ] Linear Regression
- [ ] Ridge / Lasso
- [ ] Decision Tree
- [ ] Random Forest
- [ ] Gradient Boosting (XGBoost / LightGBM)
- [ ] Evaluate each model on validation set (MAE, RMSE, R²)
- [ ] Log results for each model in `logs/`
- [ ] Select best model based on metrics
- [ ] Save best model as `.pkl` file in `models/`

## Files
- `src/model.py`
- `src/train.py`
- `logs/`
- `models/best_model.pkl`

## Definition of Done
Multiple models are trained, compared, and the best model is selected and saved.
